### ISSUE TITLE
Support node.js client? (at least DataChannel)

### ISSUE THREAD

**Issue by jonnor** [rtc-io/rtc#17]
We're using rtc.io in http://flowhub.io with http://noflojs.org to allow remote/live programming/debugging on devices from our webapp. Would like to use the same protocol for running automated test from a node.js CLI tool.

We're experimenting with the [wrtc](https://www.npmjs.com/package/wrtc) node.js library to support this. Would you be open to have support for the rtc.io offer-exchange for node.js clients in rtc itself? Or is it something that should stay separately?

**Comment by DamonOehlman** [comment]
@jonnor - Very nice.  Signing up for an opensource account to check it out :)

So I think what you want to do should already work thanks to the [`rtc-plugin-node`](https://github.com/rtc-io/rtc-plugin-node) package.  Are you using the `rtc` package or something a little lower level like [`rtc-quickconnect`](https://github.com/rtc-io/rtc-quickconnect).  Either way, it's doable but initialization will be slightly different.

The [example usage for `rtc-plugin-node`](https://github.com/rtc-io/rtc-plugin-node#example-usage) shows how to initialize the plugin when using quickconnect, and if you are using `rtc` then you should be able to specify a `plugins` option that get's passed into quickconnect which it uses internally.

Just going to take a few moments now to make sure the `rtc-plugin-node` package is up-to-date with the latest `wrtc` package...

**Comment by jonnor** [comment]
@DamonOehlman thanks for quick and comprehensive response. It does indeed look like `rtc-plugin-node` does the right thing already. So I think we can close this bug.

However, we found that for our immediate usecase (running tests against JS in headless browser) WebRTC was not as suitable as thought: neither PhantomJS 1 nor 2 has support... We still want to have it, but it got bumped down in priority a bit.

**Comment by DamonOehlman** [comment]
Yeah I've been known to rant on how useless Phantom is for any kind of actual headless testing. Unfortunately it no longer represents reality (well maybe Safari, but that's about it) as unless it has recently upgraded to use Blink for it's rendering engine it doesn't represent what is available in Chrome.

I'm not sure if it's of use but I did write a little helper tool for getting multiplatform JS tests working on travis:

https://github.com/DamonOehlman/travis-multirunner

It's reasonably well documented so if you are looking to use travis you might be able to get up and running pretty quickly.   It's used in quite a few of the rtc.io packages and also the [webrtc/adapter](https://github.com/webrtc/adapter) project actually started using it a little while ago also and the maintainers there have helped improve it quite a bit :)
